bot/cogs/snakes.py

```diff
@@ -15,31 +43,99 @@ class Snakes:
+            items = kwarg.split("=")
+            value = items[1]
+        except (IndexError, AttributeError):
+            value = kwarg
+
+        try:
+            parsed = ast.literal_eval(value)
+        except (NameError, ValueError):
+            parsed = value
+
+        return parsed
+
+    @staticmethod
+    def snake_url(name: str):
+        """Get the URL of a snake"""
+
+        def format_url(text: str):
+            """Get the full URL with that snake :D"""
+            return BASEURL.format(urllib.parse.quote_plus(text))
+
+        # Check if the snake name is valid
+        if name.upper() in [name.upper() for name in SNAKES]:
+            return format_url(name)
 
-        The information includes the name of the snake, a picture of the snake, and various other pieces of info.
-        What information you get for the snake is up to you. Be creative!
+        # Get the most similar name if a match wasn't found
+        return difflib.get_close_matches(name, SNAKES, n=5, cutoff=0)
 
-        If "python" is given as the snake name, you should return information about the programming language, but with
-        all the information you'd provide for a real snake. Try to have some fun with this!
+    @staticmethod
+    async def fetch(session, url: str):
+        """Fetch the contents of a URL as a json"""
+        async with async_timeout.timeout(10):
+            async with session.get(url) as response:
+                return await response.json()
 
-        :param name: Optional, the name of the snake to get information for - omit for a random snake
-        :return: A dict containing information on a snake
-        """
+    async def get_snek(self, name: str = None, autocorrect: str = None, details: List[str] = None) -> Dict[str, str]:
+        """If a name is provided, this gets a specific snake. Otherwise, it gets a random snake."""
+
+        autocorrect = self.parse_kwarg(autocorrect) or False
+        details = self.parse_kwarg(details) or []
+
+        if name is None:
+            name = random.choice(SNAKES)
+
+        if name.upper() == "PYTHON":
+            return PYTHON
+
+        # Get snake information
+        async with aiohttp.ClientSession() as session:
+            url = self.snake_url(name)
+
+            if isinstance(url, list):
+                if autocorrect:
+                    return await self.get_snek(url[0])
+
+                return {
+                    "name": "Oops!",
+                    "info": "We couldn't find that snake, but here are some with similar names:\n\n" + "\n".join(url),
+                }
+
+            # Get the content
+            response = await self.fetch(session, url)
+            page = response["query"]["pages"]
+            content = next(iter(page.values()))
+
+            # Parse the full-res image from the thumbnail
+            thumb = content.get("thumbnail", {}).get("source", FAILIMAGE)
+            image = "/".join(thumb.replace("thumb/", "").split("/")[:-1])
```

Comment:
`split()` takes another parameter for the number of times the string should be split. Take a look at that, maybe?